data-prepper-pipeline-parser/src/main/java/org/opensearch/dataprepper/pipeline/parser/transformer/TransformersFactory.java

```diff
@@ -4,84 +4,116 @@
         }
-        ClassLoader classLoader = TransformersFactory.class.getClassLoader();
-        InputStream filestream = classLoader.getResourceAsStream("rules" + "/" + pluginName + RULE_FILE_NAME_PATTERN);
-        return filestream;
-    }
 
-    public InputStream getPluginTemplateFileStream(String pluginName) {
-        if(pluginName == null || pluginName.isEmpty()){
-            throw  new RuntimeException("Transformation plugin not found");
+        try {
+            // Convert the URL to a URI, then to a Path to read the file
+            Path templatePath;
+            try {
+                templatePath = Paths.get(templateURL.toURI());
+            } catch (FileSystemNotFoundException e) {
+                // Handle the case where the file system is not accessible (e.g., in a JAR)
+                FileSystem fileSystem = FileSystems.newFileSystem(templateURL.toURI(), Collections.emptyMap());
+                templatePath = fileSystem.getPath(TEMPLATES_PATH + templateFileName);
+            }
+
+            // Return an InputStream for the found file
+            return Files.newInputStream(templatePath);
+
+        } catch (IOException | URISyntaxException e) {
+            throw new RuntimeException("Failed to load template file for plugin: " + pluginName, e);
         }
-        ClassLoader classLoader = TransformersFactory.class.getClassLoader();
-        InputStream filestream = classLoader.getResourceAsStream("templates" + "/" + pluginName + TEMPLATE_FILE_NAME_PATTERN);
-        return filestream;
     }
-
-    public PipelineTemplateModel getTemplateModel(String pluginName) {
-        String templatePath = getPluginTemplateFileLocation(pluginName);
+    public Collection<RuleStream> loadRules() {
+        List<RuleStream> ruleStreams = new ArrayList<>();
+        ClassLoader classLoader = getClass().getClassLoader();
 
         try {
-            PipelineTemplateModel pipelineTemplateModel = YAML_MAPPER.readValue(new File(templatePath), PipelineTemplateModel.class);
-            return pipelineTemplateModel;
-        } catch (FileNotFoundException e) {
-            throw new RuntimeException(e);
+            // Use ClassLoader to find all resources that match the RULES_PATH pattern
+            Enumeration<URL> rulesURLs = classLoader.getResources(RULES_PATH);
+
+            while (rulesURLs.hasMoreElements()) {
+                URL rulesURL = rulesURLs.nextElement();
+
+                try {
+                    // Convert the URL to a URI, then to a Path to read the directory contents
+                    Path rulesPath;
+                    try {
+                        rulesPath = Paths.get(rulesURL.toURI());
+                    } catch (FileSystemNotFoundException e) {
+                        // Handle the case where the file system is not accessible (e.g., in a JAR)
+                        FileSystem fileSystem = FileSystems.newFileSystem(rulesURL.toURI(), Collections.emptyMap());
```

Comment:
Is this going to work? These will always be in jar files.

Reply:
this works. It will always hit this code when dataprepper is running because it is always in jar during execution.